chainio/clients/eigenpod/reader.go

```diff
@@ -0,0 +1,74 @@
+package eigenpod
+
+import (
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/eigenpod/bindings"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/eth"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type ChainReader struct {
+	logger    logging.Logger
+	ethClient eth.HttpBackend
+	*bindings.IEigenPodCaller
+}
+
+type ManagerChainReader struct {
+	logger    logging.Logger
+	ethClient eth.HttpBackend
+	*bindings.IEigenPodManagerCaller
+}
+
+func newChainReader(
+	eigenPod *bindings.IEigenPodCaller,
+	ethClient eth.HttpBackend,
+	logger logging.Logger,
+) *ChainReader {
+	logger = logger.With(logging.ComponentKey, "eigenpod/reader")
+
+	return &ChainReader{
+		logger:          logger,
+		ethClient:       ethClient,
+		IEigenPodCaller: eigenPod,
+	}
+}
+
+func newManagerChainReader(
+	manager *bindings.IEigenPodManagerCaller,
+	ethClient eth.HttpBackend,
+	logger logging.Logger,
+) *ManagerChainReader {
+	logger = logger.With(logging.ComponentKey, "eigenpodmanager/reader")
+
+	return &ManagerChainReader{
+		logger:                 logger,
+		ethClient:              ethClient,
+		IEigenPodManagerCaller: manager,
+	}
+}
+
+func NewReader(
+	address common.Address,
```

Comment:
```suggestion
	eigenpodAddress common.Address,
```
just having address is error prone I feel. Or at least add a comment explaining whta address this is

Reply:
c68b91180062786b36dc2566fb7dc393ac673d89